src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/DefaultProjectWorkspaceStateGenerator.cs

```diff
@@ -161,7 +161,7 @@ private async Task UpdateWorkspaceStateAsync(Project workspaceProject, ProjectSn
                         workspaceState = new ProjectWorkspaceState(tagHelperResolutionResult.Descriptors, csharpLanguageVersion);
                     }
                 }
-                catch (TaskCanceledException)
```

Comment:
Why not also support `TaskCanceledException`?

Reply:
`TaskCanceledException` derives from `OperationCanceledException`, so this encompasses both. Actually if you try to write it as 
```c#
catch(TaskCanceledException)
catch(OperationCanceledException)
```
an analyzer will yell at you about the inheritance being the same.

Reply:
@ryanbrandenburg we should also update the surrounding try catch to to "no-op" for the various cancellation exceptions so we don't push subsequent cancellations into activity log entries.

Reply:
And that's to protect us from cancellation in the `ReportWorkspaceStateChange` Task below?

Reply:
Yes